Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityHandMeshProvider.cs

```diff
@@ -167,16 +175,24 @@ public void UpdateHandMesh(SpatialInteractionSourceState sourceState)
                     if (neutralPoseVersion != poseVersion)
                     {
                         // Compute neutral pose
-                        Vector3[] neutralPoseVertices = new Vector3[handMeshObserver.VertexCount];
+                        if ((neutralPoseVertices == null) ||
+                            (neutralPoseVertices.Length != handMeshObserver.VertexCount))
+                        {
+                            neutralPoseVertices = new Vector3[handMeshObserver.VertexCount];
+                        }
                         HandPose neutralPose = handMeshObserver.NeutralPose;
-                        var neutralVertexAndNormals = new HandMeshVertex[handMeshObserver.VertexCount];
+                        if ((neutralVertexAndNormals == null) ||
+                            (neutralVertexAndNormals.Length != handMeshObserver.VertexCount))
+                        {
+                             neutralVertexAndNormals = new HandMeshVertex[handMeshObserver.VertexCount];
+                        }
                         HandMeshVertexState handMeshVertexState = handMeshObserver.GetVertexStateForPose(neutralPose);
                         handMeshVertexState.GetVertices(neutralVertexAndNormals);
 
-                        Parallel.For(0, handMeshObserver.VertexCount, i =>
+                        for (int i = 0; i < handMeshObserver.VertexCount; i++)
```

Comment:
+1 to removing the Parallel here.

Tested this example (as well as passing the struct by ref (ConvertToUnityVector3) vs not (ToUnityVector3)) on an HL2 and after ~24k test runs, the times all converged around (in ms, per loop for ~4300 vertices):

```
Parallel if ref | 0.656342586422317
Regular if ref | 0.153393236151599
Parallel if | 0.546767675968345
Regular if | 0.115922757184502
```

Parallel is significantly worse, and ref seems additionally worse than just `new`ing new structs.

Reply:
My test code if anybody wants to double check. I saw closer numbers in the editor, but on-device parallel becomes _way_ less performant

```c#
using System.Diagnostics;
using System.Threading.Tasks;
using UnityEngine;

public class TestArray : MonoBehaviour
{
    private const int NumTests = 4300;
    private const int NumTestRuns = 5;

    private static readonly TestStruct[] Array1 = new TestStruct[NumTests];
    private static readonly TestStruct[] Array2 = new TestStruct[NumTests];
    private static readonly TestStruct[] Array3 = new TestStruct[NumTests];
    private static readonly TestStruct[] Array4 = new TestStruct[NumTests];

    private double test1millis = 0;
    private double test2millis = 0;
    private double test3millis = 0;
    private double test4millis = 0;

    private uint runCount = 0;

    private void Update()
    {
        runCount++;
        Stopwatch stopWatch = new Stopwatch();

        for (int j = 0; j < NumTestRuns; j++)
        {
            stopWatch.Restart();
            Parallel.For(0, NumTests, i =>
            {
                Vector3 newPosition = Vector3.one;
                Vector3 newPosition2 = Vector3.one;
                newPosition2.ConvertToUnityVector3(ref newPosition);
                Array1[i].Position = newPosition;
            });
            stopWatch.Stop();
            test1millis += stopWatch.Elapsed.TotalMilliseconds;

            stopWatch.Restart();
            for (int i = 0; i < NumTests; i++)
            {
                Vector3 newPosition = Vector3.one;
                Vector3 newPosition2 = Vector3.one;
                newPosition2.ConvertToUnityVector3(ref newPosition);
                Array2[i].Position = newPosition;
            }
            stopWatch.Stop();
            test2millis += stopWatch.Elapsed.TotalMilliseconds;

            stopWatch.Restart();
            Parallel.For(0, NumTests, i =>
            {
                Vector3 newPosition = Vector3.one;
                Array3[i].Position = newPosition.ToUnityVector3();
            });
            stopWatch.Stop();
            test3millis += stopWatch.Elapsed.TotalMilliseconds;

            stopWatch.Restart();
            for (int i = 0; i < NumTests; i++)
            {
                Vector3 newPosition = Vector3.one;
                Array4[i].Position = newPosition.ToUnityVector3();
            }
            stopWatch.Stop();
            test4millis += stopWatch.Elapsed.TotalMilliseconds;
        }

        UnityEngine.Debug.Log("Parallel if ref | " + (test1millis / (NumTestRuns * runCount)));
        UnityEngine.Debug.Log("Regular if ref | " + (test2millis / (NumTestRuns * runCount)));
        UnityEngine.Debug.Log("Parallel if | " + (test3millis / (NumTestRuns * runCount)));
        UnityEngine.Debug.Log("Regular if | " + (test4millis / (NumTestRuns * runCount)));
        UnityEngine.Debug.Log(NumTestRuns * runCount);
    }
}

public static class Test
{
    public static void ConvertToUnityVector3(this Vector3 source, ref Vector3 target)
    {
        target.x = source.x;
        target.y = source.y;
        target.z = -source.z;
    }

    public static Vector3 ToUnityVector3(this Vector3 vector)
    {
        return new Vector3(vector.x, vector.y, -vector.z);
    }
}
```